packages/pds/src/api/com/atproto/server/getServiceAuth.ts

```diff
@@ -1,16 +1,34 @@
-import { createServiceJwt } from '@atproto/xrpc-server'
+import { InvalidRequestError, createServiceJwt } from '@atproto/xrpc-server'
 import AppContext from '../../../../context'
 import { Server } from '../../../../lexicon'
+import { MINUTE } from '@atproto/common'
 
 export default function (server: Server, ctx: AppContext) {
   server.com.atproto.server.getServiceAuth({
     auth: ctx.authVerifier.accessPrivileged(),
     handler: async ({ params, auth }) => {
       const did = auth.credentials.did
       const keypair = await ctx.actorStore.keypair(did)
+      const exp = params.exp ? params.exp * 1000 : undefined
+      if (exp) {
+        const diff = exp - Date.now()
+        if (diff < 0) {
+          throw new InvalidRequestError(
+            'expiration is in past',
+            'BadExpiration',
+          )
```

Comment:
One trick about using `exp` versus something like a duration is that it gets another clock in the mix (our server plus the requester), plus whatever latency between the servers.  I do like that it's a passthrough to the jwt claim, though.  Not strong feels/suggestions here, just an observation.